### PR TITLE
Add Floating IP support.

### PR DESCRIPTION
**Pull request by andrewsomething**

### CODE REVIEW

**Comment by andrewsomething** [comment]
@aybabtme :eyes: ?

**Comment by aybabtme** [comment]
ohh just saw that

File: client_test.go [hunk 1]
```diff
@@ -143,6 +160,11 @@ func TestMetadata(t *testing.T) {
   "interfaces": {
     "public": [
       {
+        "anchor_ipv4": {
+          "gateway": "10.17.0.1",
```
Comment (aybabtme) [inline client_test.go]: sanitize this to be random `192.168.x.x` crap

File: client_test.go [hunk 2]
```diff
@@ -163,6 +185,12 @@ func TestMetadata(t *testing.T) {
       "8.8.8.8",
       "8.8.4.4"
     ]
+  },
+  "floating_ip": {
+    "ipv4": {
+      "ip_address": "45.55.98.102",
```
Comment (aybabtme) [inline client_test.go]: same same

File: client_test.go [hunk 3]
```diff
@@ -129,6 +129,23 @@ func TestNameservers(t *testing.T) {
 	})
 }
 
+func TestFloatingIPv4Active(t *testing.T) {
```
Comment (aybabtme) [inline client_test.go]: add tests where `resp` is not `true`:

```go
resp = "random shit"
resp = "false"
resp = ""
```

**Comment by aybabtme** [comment]
@andrewsomething lgtm after those comments, ping me again when you're ready!

**Comment by andrewsomething** [comment]
@aybabtme Updated. Thanks for the review!

File: client_test.go [hunk 4]
```diff
@@ -143,6 +180,11 @@ func TestMetadata(t *testing.T) {
   "interfaces": {
     "public": [
       {
+        "anchor_ipv4": {
+          "gateway": "192.168.0.100",
+          "netmask": "255.255.0.0",
+          "ip_address": "10.17.0.229"
```
Comment (mdlayher) [inline client_test.go]: Sanitize this too!

**Comment by mdlayher** [comment]
:+1: after last sanitation change.

**Comment by andrewsomething** [comment]
Opps. Should be good to go now.

**Comment by mdlayher** [comment]
Mind squashing into one commit too?  Sorry for the trouble!

**Comment by andrewsomething** [comment]
:+1:

**Comment by aybabtme** [comment]
thanks @andrewsomething !

**Pull request merged**